src/sqlfluff/core/templaters/jinja.py

```diff
@@ -339,18 +343,27 @@ def _slice_template(cls, in_str: str) -> Iterator[RawFileSlice]:
             # parts of the tag at a time.
             if elem_type.endswith("_end") or elem_type == "raw_begin":
                 block_type = block_types[elem_type]
+                block_subtype = None
                 # Handle starts and ends of blocks
                 if block_type == "block":
                     # Trim off the brackets and then the whitespace
-                    trimmed_content = str_buff[2:-2].strip()
```

Comment:
The old code assumed Jinja block markers were always two characters at beginning (`{%`) and end (`%}`). Updating it to use regex that knows about the optional `+` or `-` for whitespace control.